apis/v1alpha2/shared_types.go

```diff
@@ -86,6 +86,9 @@ type ParentRef struct {
 	SectionName *SectionName `json:"sectionName,omitempty"`
 }
 
+// ParentRef is a backwards compatibility alias for ParentReference.
+type ParentRef = ParentReference
```

Comment:
I think this is upsetting the code generation?

Reply:
Yup

> deepcopy-gen generates code for both the type and its alias, which is deeply wrong

Reply:
I think not having this type of backward compatibility with Go APIs like this is reasonable. Larger changes may pose problems but this is okay.
Sensitive area so would like to hear others.

Reply:
+1, I am not too concerned with go compatibility. Especially when its a trivial change

Reply:
Agreed, this seems fine. Upstream kubernetes has also not been concerned about go compat historically.

Reply:
Sounds great! I removed the compatibility alias, so let's see what CI says about that.